profiler/profiler_test.go

```diff
@@ -505,6 +484,110 @@ func assertContainsCPUProfileRateLog(t *testing.T, traceData []byte, cpuProfileR
 	assert.True(t, bytes.Contains(traceData, []byte(fmt.Sprintf("%d", cpuProfileRate))))
 }
 
+func sliceContains[T comparable](haystack []T, needle T) bool {
+	for _, s := range haystack {
+		if s == needle {
+			return true
+		}
+	}
+	return false
+}
```

Comment:
NIT: We could consider using exp/slices going forward. https://pkg.go.dev/golang.org/x/exp/slices#Contains

Reply:
Yeah, I considered that but decided against it because that package recently introduced a [breaking change](https://go-review.googlesource.com/c/exp/+/511895) (not in that function, but in a related one) that bit us during a previous release. I'd rather not depend on the package directly if we can help it, and we can use the standard lib slices package once we drop 1.20 support.